sencha-workspace/packages/slate-cbl/src/view/student/skill/OverviewWindowController.js

```diff
@@ -0,0 +1,86 @@
+/*jslint browser: true, undef: true *//*global Ext,Slate*/
+Ext.define('Slate.cbl.view.student.skill.OverviewWindowController', {
+    extend: 'Ext.app.ViewController',
+    alias: 'controller.slate-cbl-student-skill-overviewwindow',
+    requires: [
+//        'Slate.cbl.API',
+//
+//        'Ext.MessageBox',
+//        'Ext.window.Toast'
+    'Slate.cbl.model.Competency',
+    'Slate.cbl.model.Skill'
+    ],
+
+    config: {
+        id: 'slate-cbl-student-skill-overviewwindow', // workaround for http://www.sencha.com/forum/showthread.php?290043-5.0.1-destroying-a-view-with-ViewController-attached-disables-listen-..-handlers
+        control: {
+            '#': {
+                beforeshow: 'onBeforeWindowShow',
+                demorowclick: 'onDemoRowClick'
+            },
+            'combobox[reference=skillCombo]': {
+                change: 'onSkillChange'
+            }
+        }
+    },
+
+    // workaround for http://www.sencha.com/forum/showthread.php?290043-5.0.1-destroying-a-view-with-ViewController-attached-disables-listen-..-handlers
+    applyId: function(id) {
+        return Ext.id(null, id);
+    },
+
+    onBeforeWindowShow: function(overviewWindow) {
+        var me = this,
+            competencyId = overviewWindow.getCompetency(),
+            student = overviewWindow.getStudent(),
+            skillsCombo = me.lookupReference('skillCombo'),
+            skillStore = skillsCombo.getStore(),
+            skillId = overviewWindow.getSkill();
+    
+        if (skillId && competencyId) {
+            
+            if(!skillStore.isLoaded()) {
+                Slate.cbl.API.getSkills(competencyId, function(response){
+                    skillStore.loadRawData(response.data);
+
+                    skillsCombo.setValue(skillStore.findRecord('ID', skillId));
+                    me.syncDemonstrationsTable();
+                });
+
+            } else {
+                me.syncDemonstrationsTable();
+            }          
+        }
+    },
+    
+    onDemoRowClick: function(overviewWindow, ev, targetEl) {
+        targetEl.next('.skill-grid-demo-detail-row').toggleCls('is-expanded');
+        overviewWindow.doLayout();
+    },
+
+    onSkillChange: function(skillCombo, skill) {
+        this.syncDemonstrationsTable();
```

Comment:
No, that would be evaluating `this` during Ext.define which is the prototype stage before an instance actually exists, it would be equal to `window`